### PR TITLE
Changes Share button text and fixes error on desktop mode

### PR DESCRIPTION
**Pull request by prateekbh**
This PR

- Fixes desktop error for pages without backgrounds.
- Fades between backgrounds(current implementation does not allow going from image background to blank background)
- Fixes styles on `Share` button

cc: @alanorozco @newmuis @choumx

### CODE REVIEW

File: examples/amp-story/backgrounds.html [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<!doctype html>
```
Comment (newmuis) [inline examples/amp-story/backgrounds.html]: nit: maybe rename this to desktop-backgrounds.html?

File: extensions/amp-story/0.1/amp-story.js [hunk 2]
```diff
@@ -717,8 +717,14 @@ export class AmpStory extends AMP.BaseElement {
    * @return {string} The URL of the background resource
```
Comment (newmuis) [inline extensions/amp-story/0.1/amp-story.js]: Seems like this should actually be `{?string}`

File: extensions/amp-story/0.1/background.js [hunk 3]
```diff
@@ -66,10 +67,20 @@ export class AmpStoryBackground {
...
+  }
+
+  rotateActiveBackground_() {
     const newHidden = this.active_;
     this.active_ = this.hidden_;
     this.hidden_ = newHidden;
-    this.container_.appendChild(this.active_);
+    this.active_.classList.add('active');
+    this.hidden_.classList.remove('active');
+    //this.container_.appendChild(this.active_);
```
Comment (newmuis) [inline extensions/amp-story/0.1/background.js]: nit: remove commented out line if it's not needed

File: examples/amp-story/backgrounds.html (hunk 1 above)
Comment (prateekbh) [reply examples/amp-story/backgrounds.html]: Done

File: extensions/amp-story/0.1/amp-story.js (hunk 2 above)
Comment (prateekbh) [reply extensions/amp-story/0.1/amp-story.js]: Done

File: extensions/amp-story/0.1/background.js (hunk 3 above)
Comment (prateekbh) [reply extensions/amp-story/0.1/background.js]: Done

File: extensions/amp-story/0.1/amp-story.js [hunk 4]
```diff
@@ -717,14 +717,15 @@ export class AmpStory extends AMP.BaseElement {
...
-        scopedQuerySelector(pageElement, '[template="fill"]');
+    let fillElement;
 
-    if (!fillElement) {
-      return;
+    try {
+      fillElement = dev().assertElement(
+          scopedQuerySelector(pageElement, '[template="fill"]'));
+    } catch (e) {
+      return '';
```
Comment (newmuis) [inline extensions/amp-story/0.1/amp-story.js]: Why not just remove the `dev().assertElement`?  By default, `scopedQuerySelector` should return `null` if the element wasn't found.

File: extensions/amp-story/0.1/background.js [hunk 5]
```diff
@@ -57,19 +58,35 @@ export class AmpStoryBackground {
   }
 
   /**
-   * Update the background and move the previous background behind the new one.
-   * @param {string} newUrl
+   * Update the background with new background image URL.
+   * @param {string|undefined} newUrl
```
Comment (newmuis) [inline extensions/amp-story/0.1/background.js]: Shouldn't this be `{?string}`?

File: extensions/amp-story/0.1/amp-story.js (hunk 4 above)
Comment (prateekbh) [reply extensions/amp-story/0.1/amp-story.js]: its more or less about annotations, I tried that but `scopedQuerySelector` expects an element but closure complains complains that the passed argument might be null

Comment (newmuis) [reply extensions/amp-story/0.1/amp-story.js]: Then maybe we can do:

```javascript
  const fillElementOrNull = scopedQuerySelector(pageElement, '[template="fill"]');

  if (!fillElement) {
    return null;
  }

  const fillElement = dev().assertElement(fillElementOrNull);
  const fillPosterElement = scopedQuerySelector(fillElement, '[poster]');
  const srcElement = scopedQuerySelector(fillElement, '[src]');
```

Or, just cast it:


```javascript
  const fillElement = scopedQuerySelector(pageElement, '[template="fill"]');

  if (!fillElement) {
    return null;
  }

  const fillPosterElement = scopedQuerySelector(/** @type {!Element} */ (fillElement), '[poster]');
  const srcElement = scopedQuerySelector(/** @type {!Element} */ (fillElement), '[src]');
```

File: extensions/amp-story/0.1/background.js (hunk 5 above)
Comment (prateekbh) [reply extensions/amp-story/0.1/background.js]: Done

File: extensions/amp-story/0.1/amp-story.js (hunk 4 above)
Comment (prateekbh) [reply extensions/amp-story/0.1/amp-story.js]: Done

**Comment by prateekbh** [comment]
is everyone struggling with integration test or is it just me?

**Pull request merged**